### PR TITLE
Possibly fixing bugs in #175 and #205

### PR DESCRIPTION
**Pull request by rblake-llnl**
* Use correct flags when using Fortran and CUDA.
* If Fortran and CXX flags are the same, don't use a generator expression.

### CODE REVIEW

File: cmake/SetupCompilerOptions.cmake [hunk 1]
```diff
@@ -121,12 +121,18 @@ message(STATUS "OpenMP Support is ${ENABLE_OPENMP}")
 if(ENABLE_OPENMP)
     find_package(OpenMP REQUIRED)
     message(STATUS "OpenMP CXX Flags: ${OpenMP_CXX_FLAGS}")
-    
+
+    if (ENABLE_FORTRAN AND NOT OpenMP_CXX_FLAGS STREQUAL OpenMP_Fortran_FLAGS)
```
Comment (kennyweiss) [inline cmake/SetupCompilerOptions.cmake]: Perhaps it would be a good idea to add a line about why this is here/necessary?

Comment (rblake-llnl) [reply cmake/SetupCompilerOptions.cmake]: Yep, added a comment.

**Comment by white238** [comment]
Not sure why this sat for so long but I will update the branch and as long as it passes, I will merge.

**Comment by cyrush** [comment]
@white238 thanks, I was just going to ask about getting this merged!

**Pull request merged**